sysdeps/nacl/sigprocmask.c

```diff
@@ -1,66 +1,41 @@
-/* Copyright (C) 1997-2001,2003,2004,2006 Free Software Foundation, Inc.
-   This file is part of the GNU C Library.
-
-   The GNU C Library is free software; you can redistribute it and/or
-   modify it under the terms of the GNU Lesser General Public
-   License as published by the Free Software Foundation; either
-   version 2.1 of the License, or (at your option) any later version.
-
-   The GNU C Library is distributed in the hope that it will be useful,
-   but WITHOUT ANY WARRANTY; without even the implied warranty of
-   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
-   Lesser General Public License for more details.
-
-   You should have received a copy of the GNU Lesser General Public
-   License along with the GNU C Library; if not, write to the Free
-   Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
-   02111-1307 USA.  */
-
 #include <errno.h>
 #include <signal.h>
-#include <string.h>	/* Neede for string function builtin redirection.  */
-#include <unistd.h>
+#include <stddef.h>
 
-#include <sysdep.h>
-#include <sys/syscall.h>
-#include <bp-checks.h>
+#include <irt_syscalls.h>
+#include <nacl_signal.h>
 
-#include <kernel-features.h>
+void __remove_unwanted_signals(uint64_t *nacl_set) {
+  *nacl_set &= ~(1 << (SIGKILL-1));
+  *nacl_set &= ~(1 << (SIGSTOP-1));
+}
 
+/* Get and/or change the set of blocked signals.  */
+int __sigprocmask (int how, const sigset_t *set, sigset_t *oldset)
+{
+  uint64_t nacl_set = 0;
+  uint64_t nacl_oldset = 0;
+  int result;
 
-/* The variable is shared between all wrappers around signal handling
-   functions which have RT equivalents.  The definition is in sigaction.c.  */
-extern int __libc_missing_rt_sigs;
+  __sigset_t_to_uint(set, &nacl_set);
+  __sigset_t_to_uint(oldset, &nacl_oldset);
```

Comment:
some weird whitespace/indentation here

Reply:
Those are just two spaces. I think git didn't highlight it because there are also two spaces in this line in the previous version of this file.